src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-edit-form/directions-institution-hierarchies-edit-form.component.ts

```diff
@@ -0,0 +1,163 @@
+import { Component, Inject, AfterViewInit, ViewChild, ElementRef } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { MatOption } from '@angular/material/core';
+import { MatSelect } from '@angular/material/select';
+import { Router } from '@angular/router';
+import { Select, Store } from '@ngxs/store';
+import { Observable, forkJoin, asyncScheduler } from 'rxjs';
+import { InstituitionHierarchy } from '../../../../../shared/models/institution.model';
+import { GetAllInstitutionsHierarchy, GetDirections, UpdateInstitutionHierarchy } from '../../../../../shared/store/meta-data.actions';
+import { MetaDataState } from '../../../../../shared/store/meta-data.state';
+import { Direction } from '../../../../../shared/models/category.model';
+import { DataItem } from '../../../../../shared/models/item.model';
+import { EditInsHierarchyModel } from '../edit-ins-hierarchy-model';
+
+@Component({
+  selector: 'app-directions-institution-hierarchies-edit-form',
+  templateUrl: './directions-institution-hierarchies-edit-form.component.html',
+  styleUrls: ['./directions-institution-hierarchies-edit-form.component.scss']
+})
+export class DirectionsInstitutionHierarchiesEditFormComponent implements AfterViewInit {
+  @ViewChild('editForm') editForm: ElementRef;
+  @ViewChild('select') select: MatSelect;
+
+  @Select(MetaDataState.directions) directions$: Observable<Direction[]>;
+
+  editFormTitle: string = 'Редагувати напрямок';
+  editFormSubtitle: string = 'Редагуйте новий напрямок';
+  ministryLabel: string = 'Підпорядкування (Міністерство)';
+  warningMsg: string = 'Зверніть увагу, що після зміни назви напрямку для всіх \n' +
+    'інших залежних профілів також поміняється назва. ';
+  userDirectionsLabel: string = 'Напрямки для користувача';
```

Comment:
Just to clarify. We don't currently support translation, do you plan to add it in the future?

Reply:
if there is requirment, I will do that

Reply:
That would be good because we need to support multilingualism